core/server/data/slack/slack.js

```diff
@@ -0,0 +1,122 @@
+var https           = require('https'),
+    errors          = require('../../errors'),
+    url             = require('url'),
+    Promise         = require('bluebird'),
+    config          = require('../../config'),
+    events          = require('../../events'),
+    api             = require('../../api/settings'),
+    i18n            = require('../../i18n'),
+    schema          = require('../schema').checks,
+    options,
+    req,
+    slack = {},
+    slackData = {};
+
+function getSlackSettings() {
+    return api.read({context: {internal: true}, key: 'slack'}).then(function (response) {
+        var slackSetting = response.settings[0].value;
+
+        try {
+            slackSetting = JSON.parse(slackSetting) || slackSetting;
+        } catch (e) {
+            return Promise.reject(e);
+        }
+
+        return slackSetting[0];
+    });
+}
+
+function makeRequest(reqOptions, reqPayload) {
+    req = https.request(reqOptions);
+
+    reqPayload = JSON.stringify(reqPayload);
+
+    req.write(reqPayload);
+    req.on('error', function (error) {
+        errors.logError(
+            error,
+            i18n.t('errors.data.xml.xmlrpc.pingUpdateFailed.error'),
+            i18n.t('errors.data.xml.xmlrpc.pingUpdateFailed.help', {url: 'http://support.ghost.org'})
+        );
+    });
+
+    req.end();
+}
+
+function ping(post, sendTest) {
+    var message;
+    // If this is a post, we want to send the link of the post
+    // if (schema.isPost(post)) {
+    //
+    // } else {
+    //     // Assume a message
+    // }
+
+    // @TODO use schema.isPost as shown above (will need test updates)
+    if (post.slug) {
+        message = config.urlFor('post', {post: post}, true);
+    } else {
+        message = post.message;
+    }
+
+    return getSlackSettings().then(function (slackSettings) {
+        // Quit here if slack integration is not activated
+
+        if (slackSettings.isActive === true || sendTest === true) {
+            var icon;
+
+            // Stop right here, if there is no url or the default url provided
+            if (!slackSettings.url || slackSettings.url === '/') {
+                return;
+            }
+
+            // Only ping when not a page
+            if (post.page) {
```

Comment:
@AileenCGN 
because we listen on 'post.published', this condition is always false.
the post model differs between post or page events.

see https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L23